src/compiler/sys.ts

```diff
@@ -335,7 +335,12 @@ namespace ts {
                     data = "\uFEFF" + data;
                 }
 
-                _fs.writeFileSync(fileName, data, "utf8");
+                if (fileName === "/dev/stdout" || fileName === "-") {
+                    process.stdout.write(data, "utf8");
+                }
+                else {
+                    _fs.writeFileSync(fileName, data, "utf8");

```

Comment:
why is fs.writeFileSync not sufficient here? what can we change to make it work for all cases without the special handling?